### PR TITLE
Link to parent

### PR DESCRIPTION
**Pull request by Negi-Anurag**
https://trello.com/c/8RIBCbtd/8-10-parent-and-subsidiaries-subsidiary-linking-to-parent-request-modal

### CODE REVIEW

**Comment by Negi-Anurag** [comment]
> Is there any unit tests for this code?

Yes I am working on them but them.

File: server/models/classes/establishment.js [hunk 1]
```diff
@@ -1107,7 +1112,61 @@ class Establishment extends EntityValidator {
       throw new EstablishmentExceptions.EstablishmentRestoreException(null, this.uid, null, err, null);
     }
   }
+  //this method will update linkToParentRequested
+  async updateLinkToParentRequested(establishmentId) {
```
Comment (rchitranshi) [inline server/models/classes/establishment.js]: Please follow proper function commenting, which we had discussed.

File: server/models/classes/establishment.js [hunk 2]
```diff
@@ -1107,7 +1112,61 @@ class Establishment extends EntityValidator {
...
+        }
+      );
+      if (updatedEstablishment) {
+        return this._linkToParentRequested;
+      }
+    } catch (err) {
+      this._log(Establishment.LOG_ERROR, `linkToParentRequested - failed: ${err}`);
+    }
+  }
+  //This method will fetch parent id
```
Comment (rchitranshi) [inline server/models/classes/establishment.js]: Same for this.

File: server/models/classes/establishment.js [hunk 3]
```diff
@@ -1107,7 +1112,61 @@ class Establishment extends EntityValidator {
...
+  async fetchParentId(uid) {
```
Comment (rchitranshi) [inline server/models/classes/establishment.js]: Earlier you had created a function fetchParentName and now fetchParentId, no need to created two different functions, please create a common function to return all establishment details, also please check first whether we have this functionality available. As while login we are getting user's establishment details only.

**Pull request closed without merging**